test/Microsoft.SqlTools.ServiceLayer.UnitTests/Connection/ConnectionServiceTests.cs

```diff
@@ -247,6 +247,27 @@ public void CanCancelConnectRequestByDisconnecting()
             Assert.False(disconnectResult);
         }
 
+        /// <summary>
+        /// Verify that we cannot connect to the default database when no username is
+        /// provided as a parameter.
+        /// </summary>
+        [Test]
+        public async Task CantConnectWithoutUsername()
+        {
+            // Connect
+            var connectionDetails = TestObjects.GetTestConnectionDetails();
+            connectionDetails.UserName = "";
+            var connectionResult = await
+                TestObjects.GetTestConnectionService()
+                .Connect(new ConnectParams()
+                {
+                    OwnerUri = "file:///my/test/file.sql",
+                    Connection = connectionDetails
+                });
+
+            Assert.AreEqual(SR.ConnectionParamsValidateNullSqlAuth("UserName"), connectionResult.ErrorMessage);
```

Comment:
`Assert.That(connectionResult.ErrorMessage, Is.EqualTo(SR.Connection...));`

Using fluent style assertions is easier to read and generally provides better error messages so it's preferred to use them when possible.